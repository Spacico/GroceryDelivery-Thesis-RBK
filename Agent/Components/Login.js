import React, { Component } from 'react';
import {
    View,
    TextInput,
    TouchableOpacity,
    StyleSheet,
    TouchableHighlight,
    Text,
    AppRegistry
} from 'react-native';
// import Movies from './getlists';

export default class Login extends Component {
    constructor(props) {
        super(props);
        this.state = {
            changeFlag: props.changeFlag,
            agentName: '',
            password: ''
        };
    }

    onClickLogin() {
        fetch('http:192.168.1.4:5000/agentLogin', {
            method: 'POST',
            headers: {
                Accept: 'application/json',
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(this.state)
        })
            .then(res => {
                console.log('hi');
            })
            .catch(err => {
                throw err;
            });
        this.state.changeFlag('getLists');
    }
    render() {
        return (
            <View style={styles.content}>
                <View style={styles.inputContainer}>
                    <TextInput
                        underlineColorAndroid="transparent"
                        style={styles.input}
                        placeholder="agentName"
                        onChangeText={val => {
                            this.setState({ agentName: val });
                        }}
                    />
                    <TextInput
                        secureTextEntryt={true}
                        underlineColorAndroid="transparent"
                        style={styles.input}
                        placeholder="password"
                        onChangeText={val => {
                            this.setState({ password: val });
                        }}
                    />

                    <TouchableOpacity
                        style={styles.buttonContainer}
                        onPress={() => {
                            this.onClickLogin();
                        }}
                        underlayColor="blue"
                    >
                        <View
                            style={{
                                backgroundColor: 'skyblue',
                                height: 50
                            }}
                        >
                            <Text style={styles.buttonText}>Login</Text>
                        </View>
                    </TouchableOpacity>
                    <View>
                        <TouchableOpacity
                            style={styles.buttonContainer}
                            onPress={() => {
                                this.state.changeFlag('main');
                            }}
                        >
                            <Text style={styles.buttonText}>BACK</Text>
                        </TouchableOpacity>
                    </View>
                </View>
            </View>
        );
    }
}
AppRegistry.registerComponent('Login', () => Login);
const styles = StyleSheet.create({
    container: {
        flex: 1
    },
    content: {
        alignItems: 'center'
    },
    logo: {
        color: 'white',
        fontSize: 40,
        fontStyle: 'italic',
        fontWeight: 'bold',
        textShadowColor: '#252525',
        textShadowOffset: { width: 2, height: 2 },
        textShadowRadius: 15,
        marginBottom: 20
    },
    inputContainer: {
        margin: 20,
        marginBottom: 0,
        padding: 20,
        paddingBottom: 10,
        alignSelf: 'stretch',
        borderColor: '#fff',
        backgroundColor: 'rgba(255,255,255,0.2)'
    },
    input: {
        fontSize: 16,
        height: 40,
        padding: 10,
        marginBottom: 10,
        backgroundColor: 'rgba(255,255,255,1)'
    },
    buttonContainer: {
        alignSelf: 'stretch',
        margin: 20,
        padding: 20,
        backgroundColor: 'blue',
        borderWidth: 1,
        borderColor: '#fff',
        backgroundColor: 'rgba(255,255,255,0.6)'
    },
    buttonText: {
        fontSize: 16,
        fontWeight: 'bold',
        textAlign: 'center'
    }
});
