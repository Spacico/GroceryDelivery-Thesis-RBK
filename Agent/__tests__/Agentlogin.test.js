import React from 'react';
import Login from '../Components/Login';
import renderer from 'react-test-renderer';

describe('App REactNAtive Testing', () => {
    it('renders with out crashing', () => {
        const rendered = renderer.create(<Login />).toJSON();
        expect(rendered).toBeTruthy();
    });
    it('contains TextInput', () => {
        const TextInput = require('TextInput');
        const Tree = renderer.create(<TextInput />).toJSON();
        expect(Tree).toMatchSnapshot();
    });
    it('contains TouchableOpacity', () => {
        const TouchableOpacity = require('TouchableOpacity');
        const Tree = renderer.create(<TouchableOpacity />).toJSON();
        expect(Tree).toMatchSnapshot();
    });
    it('contains View', () => {
        const View = require('View');
        const Tree = renderer.create(<View />).toJSON();
        expect(Tree).toMatchSnapshot();
    });
});
