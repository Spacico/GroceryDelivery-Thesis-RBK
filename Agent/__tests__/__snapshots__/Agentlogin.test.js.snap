// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App REactNAtive Testing contains ScrollView 1`] = `
<RCTScrollView>
  <View />
</RCTScrollView>
`;

exports[`App REactNAtive Testing contains TextInput 1`] = `<TextInput />`;

exports[`App REactNAtive Testing contains TouchableOpacity 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
/>
`;

exports[`App REactNAtive Testing contains View 1`] = `<View />`;
